app/controllers/assigned_badges_controller.rb

```diff
@@ -2,7 +2,7 @@ class AssignedBadgesController < ApplicationController
   before_action :require_mentee
 
   def update
-    badge = current_user.assigned_badges.find_by(badge_id: params[:badge_id])
+    badge = current_user.assigned_badges.where(badge_id: params[:badge_id]).last
```

Comment:
@DMips Because as per current implementation, there can be multiple active assigned badges which belong to the same 'Badge' class. We want to pick the latest one for update.

Do let if any it can be done any better.

Reply:
Ah it's last not first :P Reading pr on mobile phone is always interesting ;)

Reply:
:D